fn_ioc_parser/fn_ioc_parser/components/function_ioc_parser.py

```diff
@@ -0,0 +1,106 @@
+# (c) Copyright IBM Corp. 2010, 2019. All Rights Reserved.
+# -*- coding: utf-8 -*-
+# pragma pylint: disable=unused-argument, no-self-use
+
+"""Function implementation"""
+import logging
+from resilient_circuits import ResilientComponent, function, handler, StatusMessage, FunctionResult, FunctionError
+from iocparser import IOCParser
+from fn_ioc_parser.util.ioc_parser_helper import *
+
+
+class FunctionComponent(ResilientComponent):
+    """Component that implements Resilient function 'function_ioc_parser"""
+
+    def __init__(self, opts):
+        """constructor provides access to the configuration options"""
+        super(FunctionComponent, self).__init__(opts)
+        self.options = opts.get("fn_ioc_parser", {})
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_ioc_parser", {})
+
+    @function("function_ioc_parser")
+    def _function_ioc_parser_function(self, event, *args, **kwargs):
+        """Function: A function to extract IOC's(Indicators of Compromise) from (text-based Data) Artifact and attachments."""
+        try:
+            # Get the function parameters:
+            ioc_parser_incident_id = kwargs.get("ioc_parser_incident_id")  # number
+            ioc_parser_task_id = kwargs.get("ioc_parser_task_id")  # number
+            ioc_parser_attachment_id = kwargs.get("ioc_parser_attachment_id")  # number
+            ioc_parser_artifact_id = kwargs.get("ioc_parser_artifact_id")  # number
+            ioc_parser_artifact_type = kwargs.get("ioc_parser_artifact_type")  # text
+            ioc_parser_artifact_value = kwargs.get("ioc_parser_artifact_value")  # text
+
+            log = logging.getLogger(__name__)
+            log.info("ioc_parser_incident_id: %s", ioc_parser_incident_id)
+            log.info("ioc_parser_task_id: %s", ioc_parser_task_id)
+            log.info("ioc_parser_attachment_id: %s", ioc_parser_attachment_id)
+            log.info("ioc_parser_artifact_id: %s", ioc_parser_artifact_id)
+            log.info("ioc_parser_artifact_type: %s", ioc_parser_artifact_type)
+            log.info("ioc_parser_artifact_value: %s", ioc_parser_artifact_value)
+
+            yield StatusMessage("Extracting IOCs from given Artifact/attachment data")
+
+            # Initialising the Resilient Rest client for making api calls to Resilient system
+            resilient_client = self.rest_client()
+
+            # Initialising the IOC Parser Helper class
+            IOCHelp_obj = IOCParserHelper()
+
+            if ioc_parser_artifact_id:
+                # A block to parse and download the data from Artifact
+                if ioc_parser_artifact_type.lower().strip() in ['string', 'email subject']:
```

Comment:
* This is too specific to only handle `string` and `email subject` Artifact types
* Would it make more sense to check that if artifact_value is defined and that it is of type String?

Reply:
when we attach a file to artifact then value will be having file name, in that case checking type will be failed.